modules/core/src/test/java/org/apache/ignite/internal/processors/cache/IgniteAbstractDynamicCacheStartFailTest.java

```diff
@@ -630,6 +743,55 @@ protected CacheConfiguration createCacheConfiguration(String cacheName) {
         return cfgs;
     }
 
+    /**
+     * Create list of cache configurations.
+     *
+     * @param failOnAllNodes {@code true} if cache store should be broken on all nodes.
+     * @param unluckyNode Node, where exception is raised.
+     * @param unluckyCfg Unlucky cache configuration number.
+     * @param cacheNum Number of caches.
+     * @param useFilter {@code true} if NodeFilter should be used.
+     *
+     * @return List of cache configurations.
+     */
+    protected List<CacheConfiguration> createCacheConfigsWithFailureMbServer(
+        boolean failOnAllNodes,
+        int unluckyNode,
+        int unluckyCfg,
+        int cacheNum,
+        boolean useFilter
+    ) {
+        assert unluckyCfg >= 0 && unluckyCfg < cacheNum;
+
+        final UUID uuid = ignite(unluckyNode).cluster().localNode().id();
+
+        List<CacheConfiguration> cfgs = new ArrayList<>();
+
+        for (int i = 0; i < cacheNum; ++i) {
+            CacheConfiguration cfg = new CacheConfiguration();
+
+            String cacheName = DYNAMIC_CACHE_NAME + "-" + i;
+
+            cfg.setName(cacheName);
+
+            if (i == unluckyCfg)
+                mbSrv.cache(cacheName);
+
+            if (useFilter)
+                cfg.setNodeFilter(new NodeFilter(uuid));
+
+            cfgs.add(cfg);
+        }
+
+        mbSrv.node(getTestIgniteInstanceName(unluckyNode));
+
+        if (failOnAllNodes)

```

Comment:
Multiline `if` statement should be wrapped with braces. But I think we should move this check to the previous `for` and make check `i == unluckyCfg || failOnAllNodes` with appropriate actions in true-branch.